plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -77,10 +77,9 @@ func (c *grpcClient) GetTrace(ctx context.Context, traceID model.TraceID) (*mode
 	trace := model.Trace{}
 	for received, err := stream.Recv(); err != io.EOF; received, err = stream.Recv() {
 		if err != nil {
-			if e, ok := status.FromError(err); !ok {
-				if e.Message() == spanstore.ErrTraceNotFound.Error() {
-					return nil, spanstore.ErrTraceNotFound
-				}
+			traceNotFoundStatus, _ := status.FromError(spanstore.ErrTraceNotFound)
+			if err.Error() == traceNotFoundStatus.Err().Error() {
```

Comment:
`traceNotFoundStatus.Err().Error()` prints as:
```
rpc error: code = Unknown desc = trace not found
```
I wonder if there's a better way than comparing strings. Especially suspicious is `code = Unknown`.

Reply:
Agree, pushed something that i like better.  It's still string comparison, but it is now comparing directly "trace not found".

The original error was created in a different process so I don't think there's any clean way to compare error equality.  I believe string comparison is as good as it gets here.